ecs_deployer/factories.py

```diff
@@ -1,7 +1,11 @@
-from vault import VaultManager
+from typing import Any, Dict
+from secret_managers import AWSSecretManager, VaultManager
 
 
-def build_secrets_manager(driver, secrets_manager_config):
+def build_secrets_manager(driver: str, secrets_manager_config: Dict[str, Any]):
```

Comment:
```suggestion
def build_secrets_manager(driver: str, secrets_manager_config: Dict[str, Any]) -> Union[AWSSecretManager, VaultManager]:
```

Reply:
IMHO, this one is a bit different, it should be `-> Type[SecretManager]` as they both inherit from SecretManager.

I was going to make another PR fixing all of those PR and adding more to the methods missing documentation, so I will skip this one and push the others you suggested